src/classes/RD2_ScheduleService.cls

```diff
@@ -1029,6 +1029,29 @@ public without sharing class RD2_ScheduleService {
             return isScheduled(schedule, referenceDate);
         }
 
+        /**
+        * @description Indicate if on the reference date, the schedule is currently in paused
+        * @param scheduleService RD2 Schedule Service
+        * @param referenceDate Reference Date
+        * @param schedules Recurring Donation Schedules
+        * @return Boolean
+        */
+        public Boolean isCurrentlyPaused(RD2_ScheduleService scheduleService, Date referenceDate, List<RecurringDonationSchedule__c> schedules) {
+            RecurringDonationSchedule__c nextPauseSchedule = getNextActivePauseSchedule(referenceDate, schedules);
+            
+            if (nextPauseSchedule == null) {
+                return false;
+            }
+
+            List<RD2_ScheduleService.Installment> installments = scheduleService.getVisualizedInstallments(
```

Comment:
scheduleService is required since PauseHandler is an inner class and cannot access to outer class's functions/attribute. I can either pass in the service or create a constructor for it to store the scheduleService